### PR TITLE
Upgrading to Spark 3.3.0

### PR DESCRIPTION
**Pull request by Bidek56**
Upgrading to Spark 3.3.0 with OpenJDK 17

### CODE REVIEW

**Comment by Bidek56** [comment]
@mlaccetti @LLay Do you think this PR can be merged? Thx

**Comment by mlaccetti** [comment]
@Bidek56, I don't have access to merge PRs; @Ghnuberath, do you have those permissions? If not, you will need to find someone at Uncharted to help.

**Comment by Ghnuberath** [comment]
@Bidek56 Apologies, I'm no longer with Uncharted either. 

Going to tag some folks who may be able to pick this up: @cdbethune @nkronenfeld

**Comment by nkronenfeld** [comment]
I will try to take a look today

**Comment by nkronenfeld** [comment]
We should probably bump this up to at least 3.2.2 - latest release on the 3.2 stream.
Maybe even just go straight to 3.3.0

File: services/spark-master-run [hunk 1]
```diff
@@ -1,5 +1,5 @@
-#!/usr/bin/with-contenv ash
+#!/usr/bin/with-contenv sh
 
-echo "spark.master  spark://$HOSTNAME:7077" > /opt/spark-2.4.5-bin-hadoop2.6/conf/spark-defaults.conf
+echo "spark.master  spark://$HOSTNAME:7077" > /opt/spark-3.3.0-bin-hadoop3/conf/spark-defaults.conf
```
Comment (nkronenfeld) [inline services/spark-master-run:3]: should this and line 6 use ${APACHE_SPARK_VERSION} and ${HADOOP_VERSION} as you've done above?

Comment (nkronenfeld) [reply services/spark-master-run:3]: similarly in spark-slave-run and spark-slave2-run

**Comment by nkronenfeld** [comment]
One last question - you changed from Alpine to Ubuntu - why was that?

**Comment by Bidek56** [comment]
> One last question - you changed from Alpine to Ubuntu - why was that?

1. `anapsix/alpine-java` is not longer maintained.
2. Ubuntu is a popular distro which I know works well with Spark, although the image size maybe bigger.

**Comment by nkronenfeld** [comment]
When I run this, I get a continual spew of
```
: No such file or directory sh
: No such file or directory sh
...
```

I take it you don't get this?  I'm not exactly a docker expert, so not sure what is going on here - any suggestions?

**Comment by Bidek56** [comment]
What are you running? I do the following commands using `Docker Server Version: 20.10.17` and it brings up Spark Scala

```
docker build -t uncharted/sparklet:3.3.0 .
docker run -p 8080:8080 -it uncharted/sparklet:3.3.0
```

**Comment by nkronenfeld** [comment]
the same thing, without the -p parameter... with it, I still get the same thing.
But I'm running on windows at the moment, so I don't really trust anything here.

**Comment by Bidek56** [comment]
Do you have `WSL2` installed?
What version of `Docker` and `Docker Desktop` are you running?

**Comment by nkronenfeld** [comment]
wsl2: yes
docker: 20.10.17
docker desktop 4.11.1

**Comment by Bidek56** [comment]
How about running: `docker run -it uncharted/sparklet sh` does this work?
Sorry but it works fine on my side. 
One way to figure out what is breaking is to:
1. comment out lines from the Dockerfile
2. `docker build -t uncharted/sparklet:3.3.0 .`
3. `docker run -p 8080:8080 -it uncharted/sparklet:3.3.0`

keep repeating it until it breaks.

**Comment by nkronenfeld** [comment]
ok, managed to switch to linux to test, and it works just fine.

**Review by nkronenfeld — APPROVED** [review]

**Comment by nkronenfeld** [comment]
sorry for the delay... we're working on my permissions to allow me to do so :-)

**Comment by nkronenfeld** [comment]
Done... sorry for the delay.

**Pull request merged**